### ISSUE TITLE
Store not getting resetting properly

### ISSUE THREAD

**Issue by yathink3** [teafuljs/teaful#41]
**What version of this package are you using?** 0.7.2

**What operating system, Node.js, and npm version?** windows

```
import createStore from "teaful";

const initialState = {
  username: "Aral",
  cart: {
    items: []
  }
};

export const { useStore, getStore, withStore } = createStore( initialState);

export default function DisplayAge() {
  const [age, setAge, resetAge] = useStore.age(45);

  console.log("Render DisplayAge", age);

  return (
    <div style={{ margin: "10px 0" }}>
      <div>{age}</div>
      <button onClick={() => setAge((s) => s + 1)}>Inc age</button>
      <button onClick={resetAge}>Reset age</button>
    </div>
  );
}

export default function ResetStore() {
  const [state, update, reset] = useStore();

  return (
    <>
      <button onClick={reset}>Reset all</button>
      <div>{JSON.stringify(state, undefined, 2)}</div>
    </>
  );
}
```

in this example i trying to reset the entire Store but it is creating age with defaultvalue as 45 but the answer shoud be undefined in global store

**Comment by aralroca** [comment]
@yathink3 Perhaps it's confusing and needs to be improved 🧐

The idea of `useStore.age(45)` is to init a missing store property (`age`) to `45`. So, your initial store incorporates this property. This way when resetting the age (`resetAge`) also returns the age to `45` and not `undefined`.

This is more thought because there are variables whose initial value cannot be created inside the `createStore` because they require some request to the API that is done inside the component.

Likewise, for version 0.8 we had in mind to improve the initial values, and we still don't have 100% defined the correct way to manage it. **So any proposal you can come up with is good.**.

The case we had in mind is when the whole store is defined by an API call (for example for editing a product and the product is the whole store), you can modify the changes, and reset them. So if you want to load another product here it would be good then to decide that you want to overwrite all the initial values by these new ones. And this is something you can't do right now.

In the end, it would be nice to find a way to be able to decide if you want to update the value + crush the initial value or just update it respecting the previous initial value.

**Comment by aralroca** [comment]
@yathink3  @danielart I am thinking now that maybe the reset feature carries more confusion and is not necessary and we could remove it 🤔:

- With the setValue you can already return to the property you want. Perhaps it requires to save in memory the initial values, like this:

    ```js
    const [value, setValue] = useStore.value()
    const lastValue = useRef(value)


    // After some changes:
    setValue(lastValue.current)
    ```

    Or manage it differently if you want to reset it for another purpose (what was inside the createStore):

```js
const { useStore } = createStore(initialStore)
// ...
const [store, setStore] = useStore()
// ...
setStore(initialStore)
```

- And so by we would avoid saving in memory all the initialStore and have only the store, so it would improve the memory of the applications that use Teaful, and if someone needs to reset a value can do it with the setValue.

**Comment by aralroca** [comment]
@danielart @yathink3 I pre-released my proposal to `0.8.0-canary.1`. Please review the notes of the PR https://github.com/teafuljs/teaful/pull/44

**Comment by danielart** [comment]
Agree, we were giving to teaful a responsibility that could be managed by developers and adding a point hard to explain breaking the "Easy" compromise from teaful.
